### ISSUE TITLE
ArgoCD Vault Plugin Spike (aka Secrets Spike)

### ISSUE THREAD

**Issue by nelsestu** [department-of-veterans-affairs/abd-vro#2914]
**User Story**

As a VRO Engineer, I want to identify the scope of any secret refactoring work that is necessary in order to make an informed decision about ArgoCD.

**Acceptance Criteria**

1. Identify the optimal solution for VRO's mixed content type secrets
  a. Can VRO's use of mixed content type secrets work in ArgoCD vault plugin?
2. Identify whether ArgoCD Vault Plugin allows for any configuration overhead reduction
  a. what is the potential implementation for mapping a directory of secrets 
  b. if one to one mapping is required, then how many individual secrets are we talking about?
3. Achieve a fully functioning BIP service.  The POC successfully deployed the BIP service, however, startup of the service was failing on secret inflation.

**Notes about work**
- [Tech Spec](https://docs.google.com/document/d/1sfJJWOMDEWre532rti7BYbQ_DDHt-O5CxB58sBPJ_lY/edit)
- [App of Apps Repo](https://github.com/department-of-veterans-affairs/va-abd-rrd-argocd-applications-vault)
- [Relates to ArgoCD Prototype](https://app.zenhub.com/workspaces/vro-team-6557e67173391c000e1409f3/issues/gh/department-of-veterans-affairs/abd-vro/2781)

**Comment by nelsestu** [comment]
Having presented the team with the ArgoCD POC, there was mixed response, and while the spec review seemed to indicate a "yes" we should use ArgoCD, the POC left the team with less confidence.  Following the demo, the team continued sharing feedback and concerns in slack. While inconclusive, Megan and Erik met the following monday, and resolved to try and mitigate/eliminate some of the team's concerns with a spike that would dive into secrets more deeply. As my health declined throughout the rest of last week, it complicated working on this issue.  At no point was this issue going particularly well, and while I was able to continue experimenting with the secrets retrieval mechanisms in ArgoCD, I was still struggling to resolve the mysterious "default branch" issue, which had broken part of the POC.   So finally after significant pain and multiple things still pointing towards giving up on ArgoCD, I spent some of Monday morning continuing with experiments I began over the weekend... and just prior to sprint retro, [I finally declared defeat](https://dsva.slack.com/archives/C04QLHM9LR0/p1715015243018779).
